app/controllers/concerns/api_helper.rb

```diff
@@ -238,4 +241,16 @@ def add_additional_params(action, bbb_params)
 
     final_params
   end
+
+  # early handling of server tag in default/override create settings
+  def apply_config_server_tag(params)
+    default = Rails.configuration.x.default_create_params
+    override = Rails.configuration.x.override_create_params
+    default_tenant, override_tenant = TenantSetting.defaults_and_overrides(@tenant&.id)
+
+    params[:'meta_server-tag'] = default_tenant[:'meta_server-tag'] if params[:'meta_server-tag'].nil? && !default_tenant[:'meta_server-tag'].nil?
+    params[:'meta_server-tag'] = default[:'meta_server-tag'] if params[:'meta_server-tag'].nil? && !default[:'meta_server-tag'].nil?
+    params[:'meta_server-tag'] = override[:'meta_server-tag'] unless override[:'meta_server-tag'].nil?
+    params[:'meta_server-tag'] = override_tenant[:'meta_server-tag'] unless override_tenant[:'meta_server-tag'].nil?
```

Comment:
This is a bit too complex, I'm having trouble understanding / following the different cases. Can we reformat this to something that's a bit more readable with some comments?

Reply:
If you prefer, this should be the slightly more nested, but maybe clearer, equivalent:
```
    # order of precedence: Tenant Override -> Global Override -> Param -> Tenant Default -> Global Default
    if !override_tenant[:'meta_server-tag'].nil?
      params[:'meta_server-tag'] = override_tenant[:'meta_server-tag']
    elsif !override[:'meta_server-tag'].nil?
      params[:'meta_server-tag'] = override[:'meta_server-tag']
    else
      params[:'meta_server-tag'] = default_tenant[:'meta_server-tag'] if params[:'meta_server-tag'].nil? && !default_tenant[:'meta_server-tag'].nil?
      params[:'meta_server-tag'] = default[:'meta_server-tag'] if params[:'meta_server-tag'].nil? && !default[:'meta_server-tag'].nil?
    end
 ```

 didn't try yet if Rubocop will be happy

Reply:
We probably simplify it quite a bit down to something closer to this
```
  params[:'meta_server-tag'] ||= 
    override_tenant[:'meta_server-tag'].presence || 
    override[:'meta_server-tag'].presence || 
    params[:'meta_server-tag'].presence || 
    default_tenant[:'meta_server-tag'].presence || 
    default[:'meta_server-tag'].presence
```

Reply:
> We probably simplify it quite a bit down to something closer to this
> 
> ```
>   params[:'meta_server-tag'] ||= 
>     override_tenant[:'meta_server-tag'].presence || 
>     override[:'meta_server-tag'].presence || 
>     params[:'meta_server-tag'].presence || 
>     default_tenant[:'meta_server-tag'].presence || 
>     default[:'meta_server-tag'].presence
> ```

Ruby magic to the rescue, I guess. ;-) Fine!

Reply:
OK, the ||= assignment was not quite fine, but otherwise it works.